Sources_v3/Workers/EventNotificationCenter.swift

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+class EventNotificationCenter {
+    let notificationCenter: NotificationCenter
+    let middlewares: [EventMiddleware]
+    
+    init(notificationCenter: NotificationCenter, middlewares: [EventMiddleware]) {
+        self.notificationCenter = notificationCenter
+        self.middlewares = middlewares
+    }
+    
+    func publish(_ event: Event, completion: ((Error?) -> Void)? = nil) {
```

Comment:
I'd suggest renaming this to `process`. `publish` can be misleading since not every event must be published, if some of the middlewares decide not to do it.

Reply:
done in 9ccf3e7d5bc8ffbd0021704e53abe9deed9320bb